bin/delete_lambda.py

```diff
@@ -10,9 +10,20 @@
 output = completed_process.stdout
 convert = json.loads(output.decode("utf-8").replace("'","'"))
 
-# Deletes ORCA lambda functions with the given prefix
 for sub in convert:
-    print("Deleting " + sub['FunctionName'])
-    lambda_output = sub['FunctionName']
-    delete_function = f"aws lambda delete-function --function-name {lambda_output}"
-    subprocess.run(delete_function, shell=True, capture_output=True)
+    lambda_output = sub['FunctionArn']
+    get_tags = f"aws lambda list-tags --resource {lambda_output}"
+    tags_process = subprocess.run(get_tags, shell=True, capture_output=True)
+    tag_output = tags_process.stdout
+    tag_convert = json.loads(tag_output.decode("utf-8").replace("'","'"))
+    try:
+        # Finds lambdas with the tag application set to ORCA and deletes the ORCA Lambdas
+        for tag in tag_convert:
+            if tag_convert[tag]['application'] == 'ORCA':
+                print("Deleting " + sub['FunctionName'])
+                del_lambda = sub['FunctionName']
+                delete_function = f"aws lambda delete-function --function-name {del_lambda}"
+                subprocess.run(delete_function, shell=True, capture_output=True)
+    # If the lambda with the provided prefix does not match the tag for deletion it is skipped and is not deleted
+    except KeyError:
+        pass
```

Comment:
I think if we remove `pass` and raise an exception, it will throw an error in case the deletion operation is interrupted or if tag is not present which is what we want. Otherwise, it will fail silently.
```
except KeyError as ke:
        raise ke
```

Reply:
Added that in, modified the script a bit to do some more comparisons because the way it was functioning is if it found a lambda without that `application` tag it would error out before deleting the ORCA lambdas

Reply:
Looks good